### PR TITLE
adjust for symbolic vtables

### PR DESCRIPTION
**Pull request by RalfJung**
The Miri side of https://github.com/rust-lang/rust/pull/99420

### CODE REVIEW

File: src/machine.rs [hunk 1]
```diff
@@ -620,12 +620,35 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
     ) -> InterpResult<'tcx, Pointer<Tag>> {
         let link_name = ecx.item_link_name(def_id);
         if let Some(&ptr) = ecx.machine.extern_statics.get(&link_name) {
+            // Various parts of the engine rely on `get_alloc_info` for size and alignment
```
Comment (oli-obk) [inline src/machine.rs:623]: What prompted this change? This could be done without the vtable changes, right?

Should probably add a test for it

Comment (RalfJung) [reply src/machine.rs:623]: > What prompted this change?

Me thinking carefully about extern statics due to ICEs caused by things I did when adjust for the vtable change. ;)

Yeah it's not really related to vtables.

**Pull request merged**